src/hotspot/share/gc/shenandoah/shenandoahSTWMark.cpp

```diff
@@ -99,8 +128,13 @@ void ShenandoahSTWMark::mark() {
 }
 
 void ShenandoahSTWMark::mark_roots(uint worker_id) {
-  ShenandoahInitMarkRootsClosure  init_mark(task_queues()->queue(worker_id));
-  _root_scanner.roots_do(&init_mark, worker_id);
+  if (ShenandoahStringDedup::is_enabled()) {
+    ShenandoahInitMarkRootsClosure<ENQUEUE_DEDUP>  init_mark(task_queues()->queue(worker_id));
```

Comment:
Excess space here and on the other branch, between `<ENQUEUE_DEDUP>` and `init_mark`.